commit 715525057423eeb6308d2c605ae9ec8325e43858
Author: Tomas Henzl <thenzl@redhat.com>
Date:   Thu Mar 29 17:23:46 2012 +0200

    [SCSI] aacraid: add an iounmap call to aac_src_ioremap
    
    The patch 116046127d1a3bad2853d02781ad9fee33f05e5a "[SCSI] aacraid: Added
    Sync.mode to support series 7/8/9 controllers" removed an iounmap call from
    aac_src_ioremap. Before that, the iounmap has been called twice with the same
    value (dev->base and dev->regs.src.bar0) and the iounmap complained about it
    (iounmap: bad address ...).
    
    The proper solution is a change the paremeter from bar0 to bar1.
    Fix this by adding a an iounmap(dev->regs.src.bar1) call.
    
    Signed-off-by: Tomas Henzl <thenzl@redhat.com>
    Acked-by: Achim Leubner <achim_leubner@pmc-sierra.com>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>

diff --git a/drivers/scsi/aacraid/src.c b/drivers/scsi/aacraid/src.c
index 2bee515..7628206 100644
--- a/drivers/scsi/aacraid/src.c
+++ b/drivers/scsi/aacraid/src.c
@@ -424,6 +424,8 @@ static int aac_src_deliver_message(struct fib *fib)
 static int aac_src_ioremap(struct aac_dev *dev, u32 size)
 {
 	if (!size) {
+		iounmap(dev->regs.src.bar1);
+		dev->regs.src.bar1 = NULL;
 		iounmap(dev->regs.src.bar0);
 		dev->base = dev->regs.src.bar0 = NULL;
 		return 0;
